documentation/src/main/asciidoc/mapstruct-reference-guide.asciidoc

```diff
@@ -924,6 +924,54 @@ public class CarMapperImpl implements CarMapper {
 ----
 ====
 
+[[passing-context]]
+=== Passing context or state objects to custom methods
+
+Additional _context_ or _state_ information can be passed through generated mapping methods to custom methods with `@Context` parameters. Such parameters are passed to other mapping methods, `@ObjectFactory` methods (see <<object-factories>>) or `@BeforeMapping` / `@AfterMapping` methods (see <<customizing-mappings-with-before-and-after>>) when applicable and can thus be used in custom code.
```

Comment:
I'd like to reformat that whole file with #991... It's even assigned to me already... 😁